src/api_server/src/http.rs

```diff
@@ -115,3 +117,84 @@ pub(crate) async fn attest(
 
     HttpResponse::Unauthorized().json(error_info)
 }
+
+/// GET /resource/{repository}/{type}/{tag}
+/// GET /resource/{type}/{tag}
+pub(crate) async fn resource(
+    request: HttpRequest,
+    repository: web::Data<Arc<dyn Repository + Send + Sync>>,
+    map: web::Data<SessionMap<'_>>,
+) -> HttpResponse {
+    let error_info = if let Some(cookie) = request.cookie(KBS_SESSION_ID) {
```

Comment:
Shall we change the logic here to something like following to reduce the nested conditions further?
```rust
let cookie = match request.cookie(KBS_SESSION_ID) {
    None => {
        log::error!("Missing KBS cookie");
        return HttpResponse::Unauthorized().json(kbs_error_info(ErrorInformationType::MissingCookie, ""));
    }
    Some(c) => c,
};

let locked_session = match map.sessions.read().await.get(cookie.value()) {
    None => {
        log::error!("Invalid KBS cookie {}", cookie.value());
        return HttpResponse::Unauthorized().json(kbs_error_info(ErrorInformationType::InvalidCookie, cookie.value()));
    }
    Some(ls) => ls,
};

...
```